### PR TITLE
As/address attribute selection

### PR DESCRIPTION
**Pull request by soncuteanca**
# Description
**Add initial implementation for "Address attribute selection triggers autofilll test case"**

- **Path Handling:**
  - Updated root directory path for Windows to `root_dir = Path(os.getcwd())`
- **Element Interaction:**
  - Added `autocomplete-richlistbox-element` selector.
  - Implemented `hover_over_element` to hover over elements.
  - Added `get_nth_element` to retrieve elements by index.
  - Created `get_primary_value` to extract primary value from elements.
 - **Test Implementation:**
   - Verified the primary value from the first autocomplete item matches the expected street address.

## Bugzilla bug ID

**ID:** 1895879
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1895879

## Type of change

Please delete options that are not relevant.

- [x ] New Test

# How does this resolve / make progress on that bug?
- In progress

Please describe the progress or significance with respect to the bug listed above.

# Screenshots / Explanations

Please upload any relevant media or add a relevant description with respect to the bug listed above.

# Comments / Concerns

Please add a short blurb about any comments or concerns that this change might cause.

### CODE REVIEW

File: modules/browser_object_autofill_popup.py [hunk 1]
```diff
@@ -24,3 +26,33 @@ def verify_no_popup_panel(self):
...
+        Parameters: index (str): The index of the element to retrieve (1-based).
+        Returns: WebElement: The nth element in the autocomplete list.
+        """
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            return self.wait.until(EC.visibility_of_element_located(
+                ("css selector", f".autocomplete-richlistbox .autocomplete-richlistitem:nth-child({index})")))
+
+    def get_primary_value(self, element):
+        """
+        Get the primary value from the autocomplete element.
```
Comment (sarinali) [inline modules/browser_object_autofill_popup.py:50]: For docstrings, can we follow the typical convention we already have in some other files?

File: tests/form_autofill/test_address_atribute_selection_triggers_autofill.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+@pytest.mark.parametrize("country_code", countries)
+def test_address_atribute_selection(driver: Firefox, country_code: str):
+    """
+    C122359 - This test verifies that after filling the autofill fields and saving the data, hovering over the first
+    item in the dropdown ensures that the actual value matches the expected value.
+    """
+    # Instantiate objects and open the navigation page
+    Navigation(driver).open()
+    af = AddressFill(driver).open()
```
Comment (sarinali) [inline tests/form_autofill/test_address_atribute_selection_triggers_autofill.py:20]: I think we should use more descriptive names for the variables!

**Review by sarinali — CHANGES_REQUESTED** [review]

File: modules/browser_object_autofill_popup.py (hunk 1 above)
Comment (soncuteanca) [reply modules/browser_object_autofill_popup.py]: Changes applied in my latest commit.

File: tests/form_autofill/test_address_atribute_selection_triggers_autofill.py (hunk 2 above)
Comment (soncuteanca) [reply tests/form_autofill/test_address_atribute_selection_triggers_autofill.py]: Changes applied in my latest commit.

File: modules/page_object_autofill_test_basic.py [hunk 3]
```diff
@@ -38,3 +38,34 @@ def save_information_basic(self, autofill_info: AutofillAddressBase):
...
+        term: str
+            The string to be sent to the input field
+        """
+        web_elem = self.get_element("form-field", field_name)
+        ba.clear_and_fill(web_elem, term, press_enter=False)
+
+    def click_form_button(self, field_name):
+        self.get_element("submit-button", field_name).click()
+
+    def double_click(self, name: str, *label: str):
```
Comment (sarinali) [inline modules/page_object_autofill_test_basic.py:62]: There is a pre existing double click method in `page_base.py`, all of the POMs inherit from this class so I think this method is redundant, what do you think?

**Review by sarinali — COMMENTED** [review]
the file `test_address_atribute_selection.py` is misspelled, did you mean `test_address_attribute_selection.py`?

File: modules/browser_object_autofill_popup.py [hunk 4]
```diff
@@ -24,3 +26,31 @@ def verify_no_popup_panel(self):
         with self.driver.context(self.driver.CONTEXT_CHROME):
             element = self.get_element("autofill-panel")
             self.expect_not(EC.element_to_be_clickable(element))
+
+    def hover_over_element(self, element: str):
```
[the remark this answers is not in the document]
Comment (soncuteanca) [reply modules/browser_object_autofill_popup.py]: Solved.

File: modules/browser_object_autofill_popup.py [hunk 5]
```diff
@@ -24,3 +26,31 @@ def verify_no_popup_panel(self):
...
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            self.actions.move_to_element(element).perform()
+
+    def get_nth_element(self, index: int):
```
[the remark this answers is not in the document]
Comment (soncuteanca) [reply modules/browser_object_autofill_popup.py]: This returns a WebElement, so it cannot be used in the Fluent Interface paradigm.

**Review by ben-c-at-moz — DISMISSED** [review]
Looks good, but DO NOT MERGE while I figure out conflicts with `main`

**Pull request merged**